lit/docs/guides/pipeline-guides.lit

```diff
@@ -0,0 +1,662 @@
+  may or may not work for you and your team.
+
+  \section{
+    \title{Parallelizing Get Steps in Jobs}
+
+    All jobs usually have \reference{get-step}{get steps} as their first set of
+    steps.
+
+    \codeblock{yaml}{{
+    jobs:
+    - name: awesome-job
+      plan:
+      - get: cool-code
+      - get: funny-binary
+      - get: the-weather
+      - task: business-stuff
+    }}
+
+    To reduce the waiting time to the length of longest running get step, put all
+    get steps under an \reference{in-parallel-step}.
+
+    \codeblock{yaml}{{
+    jobs:
+    - name: awesome-job
+      plan:
+      - in_parallel:
+        - get: cool-code
+        - get: funny-binary
+        - get: the-weather
+      - task: business-stuff
+    }}
+  }
+
+  \section{
+    \title{Specify Inputs for Put Steps}
+
+    By default \reference{put-step}'s have all artifacts from a job mounted in
+    their resource container. This can result in long initialization times for
+    put steps. It's likely that a \reference{put-step} only needs a subset of
+    all available artifacts generated throughout the job.
+
+    There are two ways to specify which artifacts to send to a
+    \reference{put-step}. You can specify \code{detect} as the
+    \reference{schema.step.put-step.inputs} or you can pass in an exact list of
+    all artifacts the \reference{put-step} needs.
+
+    \codeblock{yaml}{{
+    jobs:
+    - name: the-job
+      plan: # Get some artifacts
+      - get: apples
+      - get: apple-basket
+      - get: monkeys
+      # using detect
+      - put: apples-in-basket
+        input: detect
+        params:
+          apples-location: apples/location # matches the first get step
+          basket: apple-basket # matches the second get step
+      # or you can specify the exact inputs needed
+      - put: apples-in-basket
+        input: [apples, apple-basket]
+        params:
+          apples-location: apples/location
+          basket: apple-basket
+    }}
+
+  }
+
+  \section{
+    \title{Putting Task Config's in Files}
+
+    A lot of the pipeline examples that you will find on this site and in
+    resource repos will embed a \reference{schema.step.task-step.config} directly
+    in the pipeline. This is a nice way of clearly seeing what inputs/outputs the
+    task uses. Tasks are usually designed to be used in multiple places, maybe
+    with slightly different configuration. To support this scenario, most users
+    store task configs in files instead of embedding the config directly in the
+    pipeline.
+
+    Here's what this looks like in practice.
+
+    \titled-codeblock{todays-date.yml, stored in some git repo under a folder named "task"}{yaml}{{
+    platform: linux
+
+    image_resource: # define a default image for the task to use
+      type: registry-image
+      source:
+        repository: busybox
+
+    run:
+      path: date
+      args: ["+%Y-%m-%d"]
+    }}
+    Using the task in a pipeline.
```

Comment:
```suggestion

    Using the task in a pipeline.
```

I was thinking an extra line break before that line might look nice here, what do you think?

![Screen Shot 2021-05-18 at 3 01 50 PM](https://user-images.githubusercontent.com/19290703/118708673-00e26780-b7ea-11eb-9ed8-985ddab0191d.png)